packages/react/src/components/presentation/SignIn/component-driven/SignIn.tsx

```diff
@@ -186,13 +186,20 @@ const SignIn: FC<SignInProps> = ({className, size = 'medium', onSuccess, onError
 
   /**
    * Initialize the authentication flow.
+   * Priority: flowId > applicationId (from context) > applicationId (from URL)
    */
   const initializeFlow = async (): Promise<void> => {
-    const applicationIdFromUrl: string = new URL(window.location.href).searchParams.get('applicationId');
+    const urlParams = new URL(window.location.href).searchParams;
+    const flowIdFromUrl = urlParams.get('flowId');
+    const applicationIdFromUrl = urlParams.get('applicationId');
```

Comment:
Let's add types for these as well
```suggestion
    const applicationIdFromUrl: string = urlParams.get('applicationId');
```